### ISSUE TITLE
[expansion panel] fix accessibility issues caused by nested interactive elements

### ISSUE THREAD

**Issue by samrichardsontylertech** [tyler-technologies-oss/forge#247]
**Describe the bug:**
The expansion panel header has a hardcoded `role` of `button`, which renders nested interactive elements inaccessible. Including buttons with extra functionality or more complex text content within an expansion panel header is a common pattern which may require consumers to handle certain accessibility features themselves but is not possible in the current component.

**To Reproduce:**
1. Place a button within an expansion panel header
2. Run an accessibility audit tool such as axe

**Expected behavior:**
It should be possible to place arbitrary content in an expansion panel header without rendering it inaccessible.
